extension/app/ts/background/windows/personalSign.ts

```diff
@@ -30,37 +30,63 @@ function getAddressMetadataForEIP2612Message(message: EIP2612Message, addressInf
 	]
 }
 
+function getAddressMetadataForPermitMessage(message: Permit2, addressInfos: readonly AddressInfo[] | undefined) : [string, AddressBookEntry][] {
+	return [
+		[addressString(message.message.details.token), getAddressMetaData(message.message.details.token, addressInfos)],
+		[addressString(message.message.spender), getAddressMetaData(message.message.spender, addressInfos)],
+		[addressString(message.domain.verifyingContract), getAddressMetaData(message.domain.verifyingContract, addressInfos)],
+	]
```

Comment:
Anytime you return an object or a tuple like this, it usually is a good idea to add `as const` to ensure that you are returning the narrowest type possible, rather than a broad type which is what TS will default to.
```suggestion
	return [
		[addressString(message.message.details.token), getAddressMetaData(message.message.details.token, addressInfos)],
		[addressString(message.message.spender), getAddressMetaData(message.message.spender, addressInfos)],
		[addressString(message.domain.verifyingContract), getAddressMetaData(message.domain.verifyingContract, addressInfos)],
	] as const
```

Reply:
Though, since you have an explicit return type this doesn't actually matter much.  I would remove the explicit return type personally but I think we already discussed that and you disagree.

Reply:
well here we don't really want to return a constant sized array, but a dynamic array as its stored into one